Scripts/build-phases/LintAppLocalizedStringsUsage.swift

```diff
@@ -0,0 +1,328 @@
+        case buildProductsDir = "BUILT_PRODUCTS_DIR"
+        case devDir = "DEVELOPER_DIR"
+        case sdkDir = "SDKROOT"
+        var description: String { rawValue }
+    }
+
+    /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
+    /// Represents a reference to a file contained in the project tree.
+    struct PBXFileReference: PBXReference {
+        let name: String?
+        let path: String?
+        let sourceTree: SourceTree
+    }
+
+    /// One of the many `PBXObject` types encountered in the `.pbxproj` file format.
+    /// Represents a group (aka "folder") contained in the project tree.
+    struct PBXGroup: PBXReference {
+        let name: String?
+        let path: String?
+        let sourceTree: SourceTree
+        let children: [ObjectUUID]
+    }
+
+    /// Fallback type for any unknown `PBXObject` type.
+    struct UnknownPBXObject: PBXObject {
+        let isa: String
+    }
+
+    /// Wrapper helper to decode any `PBXObject` based on the value of their `isa` field
+    @propertyWrapper
+    struct PBXObjectWrapper: Decodable, CustomDebugStringConvertible {
+        let wrappedValue: PBXObject
+
+        static let knownTypes: [PBXObject.Type] = [
+            PBXProject.self,
+            PBXGroup.self,
+            PBXFileReference.self,
+            PBXNativeTarget.self,
+            PBXSourcesBuildPhase.self,
+            PBXBuildFile.self
+        ]
+
+        init(from decoder: Decoder) throws {
+            let untypedObject = try UnknownPBXObject(from: decoder)
+            if let objectType = Self.knownTypes.first(where: { $0.isa == untypedObject.isa }) {
+                self.wrappedValue = try objectType.init(from: decoder)
+            } else {
+                self.wrappedValue = untypedObject
+            }
+        }
+        var debugDescription: String { String(describing: wrappedValue) }
+    }
+}
+
+
+
+// MARK: - Lint method
+
+/// The outcome of running our lint logic on a file
+enum LintResult { case ok, skipped, violationsFound([(line: Int, col: Int)]) }
+
+/// Lint a given file for usages of `NSLocalizedString` instead of `AppLocalizedString`
+func lint(fileAt url: URL, targetName: String) throws -> LintResult {
+    guard ["m", "swift"].contains(url.pathExtension) else { return .skipped }
+    let content = try String(contentsOf: url)
+    var lineNo = 0
+    var violations: [(line: Int, col: Int)] = []
+    content.enumerateLines { line, _ in
+        lineNo += 1
+        guard line.range(of: "\\s*//", options: .regularExpression) == nil else { return } // Skip commented lines
+        guard let range = line.range(of: "NSLocalizedString") else { return }
+
+        // Violation found, report it
+        let colNo = line.distance(from: line.startIndex, to: range.lowerBound)
+        let message = """
+Use `AppLocalizedString` instead of `NSLocalizedString` in source files that are used in the `\(targetName)` extension target. See paNNhX-nP-p2 for more info.
+"""
```

Comment:
I wrote in the description that I basically copied 1-1 the WordPress setup. This is an instance in which I didn't.

In WordPress, this is a string defined with `"..."`. I used this format here because SwiftLint (which runs as a build phase in Woo) picked this up as a line that was too long.

I originally had this as a proper multiline string

```swift
        let message = """
Use `AppLocalizedString` instead...
See paNNhx...
"""
```

But Xcode only printed the first line of the string in its error bubble...